content-cards.php

```diff
@@ -434,6 +439,61 @@ private static function get_remote_data( $url ) {
 		}
 		return $result;
 	} 
+
+    public static  function init_preview() {
+        add_action( 'print_media_templates', array( 'Content_Cards', 'print_media_templates' ) );
+        add_action( 'admin_enqueue_scripts', array( 'Content_Cards', 'scripts_preview' ), 100 );
+    }
+	public static function ajax_shortcode() {
+		// Don't sanitize shortcodes — can contain HTML kses doesn't allow (e.g. sourcecode shortcode)
+		if ( ! empty( $_POST['shortcode'] ) ) {
+			$shortcode = stripslashes( $_POST['shortcode'] );
+		} else {
+			$shortcode = null;
+		}
+		if ( isset( $_POST['post_id'] ) ) {
+			$post_id = intval( $_POST['post_id'] );
+		} else {
+			$post_id = null;
+		}
+		if ( ! empty( $post_id ) ) {
+			global $post;
+			$post = get_post( $post_id );
+			setup_postdata( $post );
+		}
+		ob_start();
+		echo do_shortcode( $shortcode );
+		wp_send_json_success( ob_get_clean() );
+	}
+
+    /**
+     * Outputs the view inside the wordpress editor.
+     */
+    public static function print_media_templates() {
+        if ( ! isset( get_current_screen()->id ) || get_current_screen()->base != 'post' )
```

Comment:
I'm pretty sure `get_post` is post-type agnostic, but I'll test it out just to be sure.